crypto/sha/asm/keccak1600-armv4.pl

```diff
@@ -947,13 +947,13 @@ sub Round {
 .type	SHA3_squeeze,%function
 .align	5
 SHA3_squeeze:
-	stmdb	sp!,{r0,r3-r11,lr} @ push 11 registers
+	stmdb	sp!,{r0,r3-r10,lr}
```

Comment:
Do you know why it doesn't save r12? It trashes this with $bsz

Reply:
According to https://github.com/ARM-software/abi-aa/blob/main/aapcs32/aapcs32.rst:

_Register r12 (IP) may be used by a linker as a scratch register between a routine and any subroutine it calls (for details, see [Use of IP by the linker](https://github.com/ARM-software/abi-aa/blob/main/aapcs32/aapcs32.rst#use-of-ip-by-the-linker)). It can also be used within a routine to hold intermediate values between subroutine calls._

So it does not need to be saved.